src/instance.rs

```diff
@@ -217,7 +217,7 @@ impl<'py, T> Bound<'py, T> {
     /// Casts this `Bound<T>` as the corresponding "GIL Ref" type.
     ///
     /// This is a helper to be used for migration from the deprecated "GIL Refs" API.
-    pub fn as_gil_ref(&'py self) -> &'py T::AsRefTarget
+    pub fn as_gil_ref(&self) -> &'py T::AsRefTarget
```

Comment:
I do think this does not have to be tied to the lifetime 'self of the `Bound`, but only the GIL lifetime 'py. Am i correct here? Otherwise the created reference can not be returned from functions.